designer/server/src/common/helpers/build-error-details.js

```diff
@@ -1,20 +1,36 @@
 /**
- * @param {ValidationErrorItem[]} errorDetails
+ * @param {ValidationError} error
  */
-export function buildErrorDetails(errorDetails) {
-  return errorDetails.reduce((errors, detail) => {
+export function buildErrorDetails(error) {
+  return error.details.reduce((errors, { context, message }) => {
+    if (!context?.key) {
+      return errors
+    }
+
     return {
-      [detail.context?.key ?? 'unknown']: {
-        message: detail.message
+      [context.key]: {
+        text: message,
+        href: `#${context.key}`
       },
       ...errors
     }
   }, /** @type {ErrorDetails} */ ({}))
 }
 
 /**
- * @typedef {import('joi').ValidationErrorItem} ValidationErrorItem
- * @typedef {Record<string, { message: string }>} ErrorDetails
+ * @param {ErrorDetails | undefined} errorDetails
+ */
+export function buildErrorList(errorDetails) {
+  if (!errorDetails) {
+    return []
+  }
+
+  return Object.entries(errorDetails).map(([, message]) => message)
+}
```

Comment:
This is a new helper and takes the object output from `buildErrorDetails()` and turns it into an [**Error summary** component `errorList`](https://design-system.service.gov.uk/components/error-summary/#options-error-summary-example--error-list) array of objects